logins-sql/src/engine.rs

```diff
@@ -0,0 +1,262 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use login::Login;
+use error::*;
+use sync::{self, Sync15StorageClient, Sync15StorageClientInit, GlobalState, KeyBundle};
+use db::LoginDb;
+use std::path::Path;
+use serde_json;
+
+#[derive(Debug)]
+pub struct SyncInfo {
```

Comment:
The goal here is to avoid having lockbox need to remember to reinitialize sync when something from `Sync15StorageClientInit` changes. Keeping a cached version of it allows us to do that. The StorageClient is basically the tokenserver token and the `reqwest::Client` instance we use. `GlobalState` is the sync state machine.

The name is weird mainly because `SyncState` is somewhat confusing given that `GlobalState` is really the state in the sync state machine. I'm open to a different name.